central/baseimage/datastore/datastore_impl.go

```diff
@@ -0,0 +1,155 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/pkg/errors"
+	baseImageStore "github.com/stackrox/rox/central/baseimage/store/postgres"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+type datastoreImpl struct {
+	storage baseImageStore.Store
+	db      postgres.DB
+}
+
+var (
+	log       = logging.LoggerForModule()
+	imagesSAC = sac.ForResource(resources.ImageAdministration)
```

Comment:
Feels like this just complicates things because users don't actually insert base images, the watcher does. Do we really need the SAC here? @c-du